core/block.ts

```diff
@@ -1164,7 +1159,10 @@ export class Block implements IASTNodeLocation, IDeletable {
   renameVarById(oldId: string, newId: string) {
     for (let i = 0, input; (input = this.inputList[i]); i++) {
       for (let j = 0, field; (field = input.fieldRow[j]); j++) {
-        if (field.referencesVariables() && oldId === field.getValue()) {
+        if (
+          field.referencesVariables() &&
```

Comment:
Since we can expected `getVariable` to return null if the field doesn't reference any variables, probably best to deprecated `referencesVariables` as part of this change as well.

@rachel-fenichel I'm not certain what our policies are around deprecating things. I think we usually just make a judgement call. How do you feel about this one?

Reply:
> Since we can expected getVariable to return null if the field doesn't reference any variables, probably best to deprecated referencesVariables as part of this change as well.

Sounds good to me. I'll wait for your decision and update accordingly.

Reply:
That sounds right. Previously we needed the check because the string could be empty if it was just a different type of field, but the extra check is unnecessary when calling `getVariable` instead of `getValue`